src/amazon-qbusiness-anonymous-mcp-server/awslabs/amazon_qbusiness_anonymous_mcp_server/server.py

```diff
@@ -0,0 +1,98 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""awslabs amazon-qbusiness-anonymous MCP Server implementation."""
+
+from awslabs.amazon_qbusiness_anonymous_mcp_server.clients import get_qbiz_client, make_query
+from mcp.server.fastmcp import FastMCP
+from pydantic import Field
+
+
+"""
+Amazon Q Business anonymous mode MCP Server
+
+This module provides a Model Context Protocol (MCP) server for interacting with Amazon Q Business created using anonymous mode.
+
+Required Environment Variables:
+    AWS_REGION: The AWS region where your Q Business application is deployed
+    QBUSINESS_APPLICATION_ID: The unique identifier for your Q Business application
+
+AWS Credentials:
+    This module requires valid AWS credentials to be configured. Credentials can be provided via:
+    - AWS CLI configuration (~/.aws/credentials)
+    - Environment variables (AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN)
+    - IAM roles (when running on EC2 or other AWS services)
+
+"""
+
+mcp = FastMCP(
+    'awslabs.amazon-qbusiness-anonymous-mcp-server',
+    instructions='Use this MCP server to query the Amazon Q Business application created using anonymous mode to get responses based on the content you have ingested in it.',
+    dependencies=['pydantic', 'loguru', 'boto3'],
+)
+
+
+@mcp.tool(name='QBusinessQueryTool')
+async def qbiz_local_query(
+    query: str = Field(
```

Comment:
Consider adding input validation for the query parameter

Reply:
There is already a check for non-null / non-empty string. What more validation do you expect?

Reply:
Changed to 
query: str = Field(
        ..., description='User query, question or request to the Amazon Q Business application'
    )
to indicate that query is a required parameter.